src/main/java/frc/trigon/robot/poseestimation/apriltagcamera/AprilTagCamera.java

```diff
@@ -38,29 +40,35 @@ public class AprilTagCamera {
     public AprilTagCamera(AprilTagCameraConstants.AprilTagCameraType aprilTagCameraType,
                           String name, Transform3d robotCenterToCamera,
                           double thetaStandardDeviationExponent,
-                          double translationStandardDeviationExponent) {
+                          double translationStandardDeviationExponent,
+                          SimCameraProperties simulationCameraProperties) {
         this.name = name;
         this.robotCenterToCamera = robotCenterToCamera;
         this.thetaStandardDeviationExponent = thetaStandardDeviationExponent;
         this.translationStandardDeviationExponent = translationStandardDeviationExponent;
 
-        if (Robot.IS_REAL)
-            aprilTagCameraIO = aprilTagCameraType.createIOFunction.apply(name);
-        else
-            aprilTagCameraIO = new AprilTagCameraIO();
+        aprilTagCameraIO = aprilTagCameraType.createIOBiFunction.apply(name, simulationCameraProperties);
+        if (CameraConstants.SHOULD_USE_CAMERA_SIMULATION && aprilTagCameraType == AprilTagCameraConstants.AprilTagCameraType.PHOTON_CAMERA)
+            aprilTagCameraIO.addSimCamera(robotCenterToCamera);
     }
 
     public void update() {
         aprilTagCameraIO.updateInputs(inputs);
 
         robotPose = calculateBestRobotPose();
         logCameraInfo();
+        if (CameraConstants.SHOULD_USE_CAMERA_SIMULATION)
+            FieldConstants.VISION_SIMULATION.update(RobotContainer.POSE_ESTIMATOR.getCurrentPose());
```

Comment:
this should just be isSimulation

Reply:
It uses a lot of resources so I think it should also be its own thing